src/pattern/Pattern.ts

```diff
@@ -38,37 +52,71 @@ export interface Pattern extends WithWhere {}
  * @group Patterns
  */
 @mixin(WithWhere)
-export class Pattern extends PatternElement<NodeRef> {
+export class Pattern extends PatternElement {
     private withLabels = true;
     private withVariable = true;
     private previous: PartialPattern | undefined;
     private properties: Record<string, Expr> | undefined;
 
-    constructor(node: NodeRef, previous?: PartialPattern) {
+    private labels: string | string[] | LabelExpr | undefined;
+
+    constructor(nodeConfig: NodeRef | NodePattern, previous?: PartialPattern) {
+        let node = nodeConfig instanceof NodeRef ? nodeConfig : nodeConfig.variable;
+        if (!node) {
+            node = new NodeRef();
+        }
         super(node);
+
+        // Emulates not having a variable if the config option is passed without one
+        if (!(nodeConfig instanceof NodeRef) && !nodeConfig.variable) {
```

Comment:
I think you can remove this and assign `this.withVariable` as `Boolean(nodeConfig.variable)` in the branch condition below